frontend/appflowy_flutter/lib/startup/tasks/generate_router.dart

```diff
@@ -0,0 +1,270 @@
+          return CustomTransitionPage(
+            child: EncryptSecretScreen(
+              user: args['userFolder'],
+              key: args['error'],
+            ),
+            transitionsBuilder: _buildFadeTransition,
+            transitionDuration: Duration(
+              milliseconds: (RouteDurations.slow.inMilliseconds).round(),
+            ),
+          );
+        },
+      ),
+      // Desktop only
+      GoRoute(
+        path: DesktopHomeScreen.routeName,
+        pageBuilder: (context, state) {
+          final args = state.extra as Map<String, dynamic>;
+          return CustomTransitionPage(
+            child: DesktopHomeScreen(
+              key: args['key'],
+              userProfile: args['userProfile'],
+              workspaceSetting: args['workspaceSetting'],
+            ),
+            transitionsBuilder: _buildFadeTransition,
+            transitionDuration: Duration(
+              milliseconds: (RouteDurations.slow.inMilliseconds).round(),
+            ),
+          );
+        },
+      ),
+      // Mobile only
+      // We use StatefulShellRoute to create a StatefulNavigationShell(ScaffoldWithNavBar) to access to multiple pages, and each page retains its own state.
+      StatefulShellRoute.indexedStack(
+        builder: (
+          BuildContext context,
+          GoRouterState state,
+          StatefulNavigationShell navigationShell,
+        ) {
+          // Return the widget that implements the custom shell (in this case
+          // using a BottomNavigationBar). The StatefulNavigationShell is passed
+          // to be able access the state of the shell and to navigate to other
+          // branches in a stateful way.
+          return MobileBottomNavigationBar(navigationShell: navigationShell);
+        },
+        branches: <StatefulShellBranch>[
+          StatefulShellBranch(
+            routes: <RouteBase>[
+              GoRoute(
+                // The screen to display as the root in the first tab of the
+                // bottom navigation bar.
+                path: MobileHomeScreen.routeName,
+                builder: (BuildContext context, GoRouterState state) {
+                  return const MobileHomeScreen();
+                },
+              ),
+            ],
+          ),
+          // TODO(yijing): implement other tabs later
+          // The following code comes from the example of StatefulShellRoute.indexedStack. I left there just for placeholder purpose. They will be updated in the future.
+          // The route branch for the second tab of the bottom navigation bar.
+          StatefulShellBranch(
+            // It's not necessary to provide a navigatorKey if it isn't also
+            // needed elsewhere. If not provided, a default key will be used.
+            routes: <RouteBase>[
+              GoRoute(
+                // The screen to display as the root in the second tab of the
+                // bottom navigation bar.
+                path: '/b',
```

Comment:
Is it just for the demonstration? If no, please use a meaningful string.

Reply:
All the routes after MobileHomeScreen are just for demonstration. I added a TODO there.